src/aiida/engine/daemon/execmanager.py

```diff
@@ -435,56 +435,109 @@ async def stash_calculation(calculation: CalcJobNode, transport: Transport) -> N
     :param transport: an already opened transport.
     """
     from aiida.common.datastructures import StashMode
-    from aiida.orm import RemoteStashFolderData
+    from aiida.orm import RemoteStashCompressedData, RemoteStashFolderData
 
     logger_extra = get_dblogger_extra(calculation)
 
     stash_options = calculation.get_option('stash')
-    stash_mode = stash_options.get('mode', StashMode.COPY.value)
+    stash_mode = stash_options.get('stash_mode')
     source_list = stash_options.get('source_list', [])
+    uuid = calculation.uuid
+    source_basepath = Path(calculation.get_remote_workdir())
 
     if not source_list:
         return
 
-    if stash_mode != StashMode.COPY.value:
-        EXEC_LOGGER.warning(f'stashing mode {stash_mode} is not implemented yet.')
+    if stash_mode not in [mode.value for mode in StashMode.__members__.values()]:
```

Comment:
In `engine/processes/calcjobs/calcjob.py` there is a `validate_stash_options` that checks the stash mode. I am not sure, but I think plumpy fires an error in this case on the creation of the process. I need to think about this.

Also what if stash mode is `None`, then this also fires right? (you retrieving it with `get` which should return `None` if not present